### ISSUE TITLE
Resolve conversation automatically ?

### ISSUE THREAD

**Issue by abal0011** [reviewdog/action-actionlint#95]
We have been using actionlint in our internal repositories at a project that we are working on. I was wondering if actionlint would be able to resolve conversations on it own through latest workflow run ?
![image](https://user-images.githubusercontent.com/32474278/211225517-24cd259e-1fd4-4a47-898b-9f0aab902e22.png)

I can see that we have pushed a new change in after the suggestion provided by actionlint through PR comments and another run of the review was also completed through github workflow run but the PR conversation does not seem to be resolved automatically. 

Should we be looking at resolving the conversation ourselves referring to the latest workflow run's output or can the conversations be resolved automatically to avoid any human errors ?

**Comment by shogo82148** [comment]
Thanks for pointing that out. You're right that reviewdog implemented a feature in [#1804](https://github.com/reviewdog/reviewdog/pull/1804) (released in v0.19.0, 2024-07-07) where the `github-pr-review` reporter automatically deletes a previously-posted comment once the corresponding finding is no longer reported.

That said, this feature didn't exist yet when this issue was filed in January 2023 — it landed about a year and a half later. More importantly, the implementation has a limitation that's directly relevant to this issue's scenario:

```go
for _, c := range g.outdatedComments {
    if ok := g.prCommentWithReply[c.GetID()]; ok {
        // Do not remove comment with replies.
        continue
    }
    ...
}
```

([service/github/github.go](https://github.com/reviewdog/reviewdog/blob/master/service/github/github.go#L224-L232))

**Comments that have replies are not deleted** — this is by design, since there could be a meaningful discussion attached to them. In a case like this one, where the actionlint comment received a reply from the PR author or another reviewer, the comment won't be deleted even after the underlying issue is fixed, and the conversation will remain unresolved.

Also, this is comment *deletion*, not the same thing as what GitHub's "Resolve conversation" button does, which marks the thread as **resolved** via a separate API. Deletion looks similar on the surface (the thread disappears from the Files changed view), but it isn't equivalent to marking the thread resolved the way the GitLab reporter does with `ResolveMergeRequestDiscussion` ([#2543](https://github.com/reviewdog/reviewdog/pull/2543)).

In summary:
- Comments with no replies → get auto-deleted, effectively resolving the issue
- Comments with replies (i.e., cases where a human reviewer engaged in discussion) → **not deleted, and the conversation stays unresolved**, so the problem reported here isn't fully solved
